### PR TITLE
Update `internbuddy.storage` package

### LINKED ISSUE #43: Update Storage to match refactored classes
* Create `JsonAdaptedCompany` to accommodate new `status` field
* Update other classes in `storage` package to use the new class 
* Update storage tests

### PR DESCRIPTION
**Pull request by uniqly**
Will resolve #43

Currently will fail CI due to missing updated classes in the `internbuddy.logic` package. (FIXED)

### CODE REVIEW

File: src/main/java/seedu/internbuddy/model/Model.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import seedu.internbuddy.model.company.Company;
 
 /**
- * The API of the Model component.
+ * The API of the ModelCompany component.
```
Comment (blanklogic) [inline src/main/java/seedu/internbuddy/model/Model.java:11]: Think might be appropriate to leave it as Model component for accuracy to the filename

File: src/main/java/seedu/internbuddy/model/ModelManager.java [hunk 2]
```diff
@@ -24,7 +24,7 @@ public class ModelManager implements Model {
     private final FilteredList<Company> filteredCompanies;
 
     /**
-     * Initializes a ModelManager with the given addressBook and userPrefs.
+     * Initializes a ModelManagerCompany with the given addressBook and userPrefs.
```
Comment (blanklogic) [inline src/main/java/seedu/internbuddy/model/ModelManager.java:27]: Same here, ModelManager for accuracy

File: src/main/java/seedu/internbuddy/storage/JsonAdaptedCompany.java [hunk 3]
```diff
@@ -15,54 +15,58 @@
...
+import seedu.internbuddy.model.company.Status;
 import seedu.internbuddy.model.tag.Tag;
 
 /**
  * Jackson-friendly version of {@link Company}.
  */
-class JsonAdaptedCompany {
-
-    public static final String MISSING_FIELD_MESSAGE_FORMAT = "company's %s field is missing!";
+public class JsonAdaptedCompany {
```
Comment (blanklogic) [inline src/main/java/seedu/internbuddy/storage/JsonAdaptedCompany.java:24]: Curious but is there a need to be public instead of package private?

File: src/main/java/seedu/internbuddy/storage/JsonAdaptedCompany.java [hunk 4]
```diff
@@ -15,54 +15,58 @@
...
     private final String name;
     private final String phone;
     private final String email;
     private final String address;
+    private final String status;
     private final List<JsonAdaptedTag> tags = new ArrayList<>();
 
     /**
-     * Constructs a {@code JsonAdaptedcompany} with the given company details.
+     * Constructs a {@link JsonAdaptedCompany} with the given person details.
```
Comment (blanklogic) [inline src/main/java/seedu/internbuddy/storage/JsonAdaptedCompany.java:35]: Why the change in the Javadoc for this constructor as well as the reference to person?

File: src/test/java/seedu/internbuddy/logic/commands/CommandTestUtil.java [hunk 5]
```diff
@@ -99,7 +104,7 @@ public static void assertCommandSuccess(Command command, Model actualModel, Stri
      * Executes the given {@code command}, confirms that <br>
      * - a {@code CommandException} is thrown <br>
      * - the CommandException message matches {@code expectedMessage} <br>
-     * - the address book, filtered company list and selected company in {@code actualModel} remain unchanged
+     * - the address book, filtered person list and selected person in {@code actualModel} remain unchanged
```
Comment (blanklogic) [inline src/test/java/seedu/internbuddy/logic/commands/CommandTestUtil.java:107]: Why the re-referencing to person here?

File: src/test/java/seedu/internbuddy/logic/commands/CommandTestUtil.java [hunk 6]
```diff
@@ -112,10 +117,10 @@ public static void assertCommandFailure(Command command, Model actualModel, Stri
         assertEquals(expectedFilteredList, actualModel.getFilteredCompanyList());
     }
     /**
-     * Updates {@code model}'s filtered list to show only the company at the given {@code targetIndex} in the
+     * Updates {@code model}'s filtered list to show only the person at the given {@code targetIndex} in the
```
Comment (blanklogic) [inline src/test/java/seedu/internbuddy/logic/commands/CommandTestUtil.java:120]: Refer to previous

**Review by blanklogic — CHANGES_REQUESTED** [review]
Have some questions as well as some changes in mind, would love your input on them.

File: src/main/java/seedu/internbuddy/storage/JsonAdaptedCompany.java (hunk 3 above)
Comment (uniqly) [reply src/main/java/seedu/internbuddy/storage/JsonAdaptedCompany.java:24]: This is the same as in the old `JsonAdaptedPerson` I think

**Review by realqijun — APPROVED** [review]
Love how you changed some names to actual company names! LGTM
